homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -133,24 +151,94 @@ def __init__(self, coordinator, device_info, room_name, shade, name):
         self._last_action_timestamp = 0
         self._scheduled_transition_update = None
-        self._current_hd_cover_position = MIN_POSITION
         if self._device_info[DEVICE_MODEL] != LEGACY_DEVICE_MODEL:
             self._attr_supported_features |= CoverEntityFeature.STOP
         self._forced_resync = None
+        # these are used to store the value to be returned in the case
+        # coordinator data is unavailable. Serve as a fallback for key missing
+        # ie shades that only send 1 position when they have 2
+        self._cached_primary = MIN_POSITION
+        self._cached_secondary = MIN_POSITION
+        self._cached_vane = MIN_POSITION
 
     @property
     def extra_state_attributes(self):
         """Return the state attributes."""
         return {STATE_ATTRIBUTE_ROOM_NAME: self._room_name}
 
+    @property
+    def position_data(self):
+        """Return the position data."""
+        return self.coordinator.data[self._shade.id][ATTR_POSITION_DATA]
+
+    def get_position(self, pos_kind):
+        """Return position data from the co-ordinator."""
+        if self.position_data.get(ATTR_POSKIND1) == pos_kind:
+            return self.position_data[ATTR_POSITION1]
+        if self.position_data.get(ATTR_POSKIND2) == pos_kind:
+            return self.position_data[ATTR_POSITION2]
+        # cache value used if data not in co-ordinator (vanes/type8+9)
+        if pos_kind == POS_KIND_PRIMARY:
+            return self._cached_primary
+        if pos_kind == POS_KIND_SECONDARY:
+            return self._cached_secondary
+        if pos_kind == POS_KIND_VANE:
+            return self._cached_vane
+
+    @property
+    def position_primary(self):
+        """Return primary shade position."""
+        return self.get_position(POS_KIND_PRIMARY)
+
+    @property
+    def position_secondary(self):
+        """Return secondary shade position."""
+        return self.get_position(POS_KIND_SECONDARY)
+
+    @property
+    def position_vane(self):
+        """Return vane position."""
+        return self.get_position(POS_KIND_VANE)
+
+    def set_position(self, value, pos_kind):
+        """Store position into coordinator and local cache."""
+        # confirm the shade attribute in case position1/2 are inverted
+        if self.position_data.get(ATTR_POSKIND1) == pos_kind:
+            self.position_data[ATTR_POSITION1] = value
+        if self.position_data.get(ATTR_POSKIND2) == pos_kind:
+            self.position_data[ATTR_POSITION2] = value
+        # always store the last copy of information we received
+        # should not be required but will limit chance of an edge case error
+        if pos_kind == POS_KIND_PRIMARY:
+            self._cached_primary = value
+        if pos_kind == POS_KIND_SECONDARY:
+            self._cached_secondary = value
+        if pos_kind == POS_KIND_VANE:
+            self._cached_vane = value
+        _LOGGER.debug("%s - Update Position: %s", self.name, self.position_data)
```

Comment:
```suggestion
        _LOGGER.debug("%s - Update Position: %s", self.name, self.position_data)
```

https://github.com/home-assistant/core/pull/62788/files#r879026282